Mathlib/Algebra/Lie/Nilpotent.lean

```diff
@@ -671,6 +708,31 @@ theorem LieModule.isNilpotent_of_top_iff :
 
 end Morphisms
 
+namespace LieModule
+
+variable (R L M)
+
+section
+
+variable [LieModule R L M]
+
+theorem isNilpotent_of_le (M₁ M₂ : LieSubmodule R L M) (h₁ : M₁ ≤ M₂) (h₂ : IsNilpotent L M₂) :
+    IsNilpotent L M₁ := by
+  let f : L →ₗ⁅R⁆ L := 1
+  let g : M₁ →ₗ[R] M₂ := Submodule.inclusion h₁
+  have hfg : ∀ x m, ⁅f x, g m⁆ = g ⁅x, m⁆ := by
+    intro x m
+    simp only [LieHom.coe_one, id_eq, f, g]
+    obtain ⟨val, property⟩ := m
+    rfl
+  have hg_inj : Function.Injective g := by
+    apply Submodule.inclusion_injective
+  exact hg_inj.lieModuleIsNilpotent hfg
```

Comment:
Sometimes `aesop` can be slow but it will crush goals like this efficiently so no need for us so we should use it:
```suggestion
  let f : L →ₗ⁅R⁆ L := LieHom.id
  let g : M₁ →ₗ[R] M₂ := Submodule.inclusion h₁
  have hfg : ∀ x m, ⁅f x, g m⁆ = g ⁅x, m⁆ := by aesop
  exact (Submodule.inclusion_injective h₁).lieModuleIsNilpotent hfg
```